lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -503,11 +503,20 @@ namespace Js
 #else
         library->AddMember(intlNativeInterfaces, Js::PropertyIds::winglob, library->GetFalse());
 
-        // when using ICU, we need to call u_init to ensure that ICU is functioning properly before allowing Intl to continue.
+#ifdef WINDOWS10_ICU
+        // when using ICU, we can call u_init to ensure that ICU is functioning properly before allowing Intl to continue.
         // u_init will cause the data file to be loaded, and if we don't have enough memory to do so, we can throw OutOfMemory here.
         // This is to protect against spurious U_MISSING_RESOURCE_ERRORs and U_FILE_ACCESS_ERRORs coming from early-lifecycle
         // functions that require ICU data.
         // See OS#16897150, OS#16896933, and others relating to bad statuses returned by GetLocaleData and IsLocaleAvailable
+        //
+        // This is only enabled for Windows 10 ICU because it seems like u_init reports U_FILE_ACCESS_ERROR when udata_setCommonData
+        // is called first, which is the case for Node-ChakraCore. As it stands right now, there is no information plumbed to
+        // ChakraCore to know whether we are building in the context of Node or not. Thus, to be safe, we can limit this only
+        // to Windows 10 ICU because we know udata_setCommonData will not be called before then (since it is not allowed). This
+        // change primarily exists to guard against OOMs in ICU being treated as unique bugs by Crawler, so it really only *needs*
+        // to be enabled when we are using Windows 10 ICU, anyways.
+        // TODO(jahorto): investigate why u_init can't be called after udata_setCommonData
```

Comment:
Not sure I understand this. `u_init` _should_ be called after `udata_setCommonData` (that's the design and what Node.js does).   The `U_FILE_ACCESS_ERROR` may be something wrong with the data/configuration. Try compiling with `-DUDATA_DEBUG=1` and make sure your terminal's scrollback buffer is cranked way up…

Reply:
Unless I am missing something, I didn't see where node called u_init in the same path as calling udata_setCommonData. so I assumed it was intentional.